sf-docker/sf-observability/docker-compose.yml

```diff
@@ -0,0 +1,214 @@
+        condition: service_started
+      obsminio_createbuckets:
+        condition: service_completed_successfully
+    volumes:
+      - ./configuration/loki:/etc/loki
+      - loki_data:/loki
+    command: -config.file=/etc/loki/loki-config.yaml
+    expose:
+      - 3100
+    networks:
+      observability:
+    restart: unless-stopped
+  
+  alloy:
+    extends:
+      service: common
+      file: docker-compose-common.yml
+    image: grafana/alloy:v1.5.0
+    profiles: [singlenode, multinode-scraper]
+    entrypoint: [ "/bin/sh","-c" ]
+    command:
+      - |
+        export HOSTNAME=$$(cat /etc/hostname)
+        /bin/alloy run --server.http.listen-addr=0.0.0.0:12345 --storage.path=/var/lib/alloy/data /etc/alloy/config.alloy
+    environment:
+      - LOKI_URL=http://loki:3100/loki/api/v1/push
+      - PROMETHEUS_URL=http://prometheus:9090/api/v1/write
+    volumes:
+      - ./configuration/alloy/config.alloy:/etc/alloy/config.alloy
+      - /var/run/docker.sock:/var/run/docker.sock:ro
+      - /etc/hostname:/etc/hostname:ro
+    ports:
+      - 12345:12345
+    networks:
+      observability:
+      sf-network:
+    restart: unless-stopped
+
+  # Jaeger tracing as configured here is solely for debug purposes. No data is stored!
+  jaeger:
+    extends:
+      service: common
+      file: docker-compose-common.yml
+    image: "jaegertracing/all-in-one:1.39.0"
+    profiles: [singlenode, multinode-scraper]
+    command: ["--memory.max-traces=5000"]
+    expose:
+      - 5778
+      - 16686
+      - 14268
+      - 14250
+      - 9411
+      - 5775
+      - 6831
+      - 6832
+    ports:
+      - "9903:16686"
+    networks:
+      sf-network:
+        aliases:
+          - jaeger
+      observability:
+    restart: unless-stopped
+  
+  nodeexporter:
+    extends:
+      service: common
+      file: docker-compose-common.yml
+    image: prom/node-exporter:v1.6.1
+    profiles: [singlenode, multinode-scraper]
+    volumes:
+      - /proc:/host/proc:ro
+      - /sys:/host/sys:ro
+      - /:/rootfs:ro
+    command:
+      - '--path.procfs=/host/proc'
+      - '--path.rootfs=/rootfs'
+      - '--path.sysfs=/host/sys'
+      - '--collector.filesystem.mount-points-exclude=^/(sys|proc|dev|host|etc)($$|/)'
+    expose:
+      - 9100
+    networks:
+      observability:
+    restart: unless-stopped
+  
+  cadvisor:
+    extends:
+      service: common
+      file: docker-compose-common.yml
+    image: gcr.io/cadvisor/cadvisor:v0.47.2
+    profiles: [singlenode, multinode-scraper]
+    volumes:
+      - /:/rootfs:ro
+      - /var/run:/var/run:rw
+      - /sys:/sys:ro
+      - /var/lib/docker/:/var/lib/docker:ro
+    expose:
+      - 8080
+    networks:
+      observability:
+    restart: unless-stopped
+  
+  prometheus:
+    extends:
+      service: common
+      file: docker-compose-common.yml
+    image: prom/prometheus:v2.47.0
+    profiles: [singlenode, multinode-aggregator]
+    volumes:
+      - ./configuration/prometheus:/etc/prometheus
```

Comment:
Please include the config file using docker compose `configs`, also make sure to specify file attributes.

Reply:
done